forge-gui/res/cardsfolder/upcoming/ace_fearless_rebel.txt

```diff
@@ -0,0 +1,13 @@
+Name:Ace, Fearless Rebel
+ManaCost:3 G
+Types:Legendary Creature Human Rebel
+PT:2/2
+T:Mode$ Attacks | ValidCard$ Card.Self | TriggerZones$ Battlefield | Execute$ TrigPutCounter | OptionalDecider$ You | TriggerDescription$ Nitro-9 — Whenever CARDNAME attacks, you may sacrifice an artifact. When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
+SVar:TrigPutCounter:AB$ PutCounter | Cost$ Sac<1/Artifact> | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1 | SubAbility$ DBImmediateTrig | TriggerDescription$ You may sacrifice an artifact. When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
+SVar:DBImmediateTrig:DB$ ImmediateTrigger | Execute$ DBFight | TriggerDescription$ When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
+SVar:DBFight:DB$ Fight | Defined$ ParentTarget | TargetMin$ 0 | TargetMax$ 1 | ValidTgts$ Creature.ControlledBy TriggeredDefendingPlayer | TgtPrompt$ Select target creature defending player controls.

```

Comment:
First half of the ability (sacrificing the artifact and gaining the counter) now works. The fight section of the ability still doesn't work.

Reply:
Try like this
 
```
SVar:DBImmediateTrig:AB$ ImmediateTrigger |  | Cost$ Sac<1/Artifact> | Execute$ TrigPutCounter | TriggerDescription$ When you do, put a +1/+1 counter on CARDNAME, then it fights up to one target creature defending player controls.
SVar:TrigPutCounter:AB$ PutCounter | Cost$ Sac<1/Artifact> | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1 | SubAbility$ DBFight
SVar:DBFight:DB$ Fight | Defined$ ParentTarget | TargetMin$ 0 | TargetMax$ 1 | ValidTgts$ Creature.ControlledBy TriggeredDefendingPlayer | TgtPrompt$ Select target creature defending player controls.
```

Reply:
This is def. advanced rules territory:
> 603.12. A resolving spell or ability may allow or instruct a player to take an action and create a triggered ability that triggers "when [a player] [does or doesn't]" take that action or "when [something happens] this way." These reflexive triggered abilities follow the rules for delayed triggered abilities (see rule 603.7), except that they're checked immediately after being created and trigger based on whether the trigger event or events occurred earlier during the resolution of the spell or ability that created them. 

1. So the order needs fixing - you pay the cost to get the additional actions.

2. The fight sub has wrong Defined, and the ValidTgts needs `Creature.ControlledBy Spawner>TriggeredDefendingPlayer` which allows you to navigate back to the original trigger and access its triggeringObjects